ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -136,7 +143,11 @@ private void doCopyRetry(FileSystem sourceFs, List<ReplChangeManager.FileInfo> s
           }
 
           // looks like some network outrage, reset the file system object and retry.
-          FileSystem.closeAllForUGI(Utils.getUGI());
+          if (proxyUser == null) {
+            FileSystem.closeAllForUGI(Utils.getUGI());
+          } else {
+            FileSystem.closeAllForUGI(proxyUser);
+          }
           sourceFs = pathList.get(0).getFileSystem(hiveConf);
           destinationFs = destination.getFileSystem(hiveConf);
```

Comment:
We need to propagate the destinationFs to caller as well as caller still points to old file system object which is invalid. As caller uses this object in a loop, it is necessary to propagate it back to caller.

Reply:
no need ..dlm or any other system calling repl load should retry in case of file system error